pandas/core/groupby/generic.py

```diff
@@ -321,13 +321,16 @@ def _aggregate_multiple_funcs(self, arg):
                 obj = copy.copy(obj)
                 obj._reset_cache()
                 obj._selection = name
-            results[name] = obj.aggregate(func)
+            results[base.OutputKey(label=name, position=idx)] = obj.aggregate(func)
 
         if any(isinstance(x, DataFrame) for x in results.values()):
             # let higher level handle
             return results
 
-        return self.obj._constructor_expanddim(results, columns=columns)
+        if not results:
+            return DataFrame()
```

Comment:
hmm is this correct? do we have tests that hit this. I would think we would have *somthing* e.g. columns even if this is empty

Reply:
also why is this not just handled in wrap_aggregated_output?

Reply:
Here's a test that hits it: `pandas/tests/groupby/aggregate/test_aggregate.py::TestNamedAggregationSeries::test_no_args_raises`

When trying to move this to `wrap_aggregated_output` I ran into #34977, so I'll try to address that first

Reply:
this still is quite fishy . if you pass en empty result to `self._wrap_aggregated_output` what do you get as `output`? I really don't like special cases like this which inevitably hide errors and make groking code way more complex.

So prefer to have _wrap_aggregated_output handle this correctly. you *may* not even need L333, its possible to pass columns to _wrap_aggregated_output

Reply:
i would just fix this, need to check if len(indexed_output)

Reply:
So,
```python
elif len(indexed_output):
    result = self.obj._constructor(
        indexed_output[0], index=index, name=columns[0]
    )
else:
    result = self.obj._constructor()
```
?

I can do that, but then I'll still have to address #34977 when the output of `_wrap_aggregated_output` is passed to `self.obj._constructor_expanddim(results, columns=columns)`. 

> its possible to pass columns to _wrap_aggregated_output

Are you sure? It seems to only take on argument (other than `self`)

Reply:
Another option would be to return `None` from `_wrap_aggregated_output`, so the return type would be `Optional[Union[Series, DataFrame]]` - would that be OK?

Reply:
@jreback would
```python
elif len(indexed_output):
    result = self.obj._constructor(
        indexed_output[0], index=index, name=columns[0]
    )
else:
    return None
```
be an acceptable solution?

Reply:
Actually,
```python
        elif not columns.empty:
            result = self.obj._constructor(
                indexed_output[0], index=index, name=columns[0]
            )
        else:
            result = self.obj._constructor_expanddim()
```
works, because
```python
pd.DataFrame(pd.DataFrame(), columns=[])
```
is allowed.

No need to modify the return types like this :)